### PR TITLE
HV-1863 HV-1900 Better (and tested) support for records

### PR DESCRIPTION
**Pull request by jrenaat**
- https://hibernate.atlassian.net/browse/HV-1863
- https://hibernate.atlassian.net/browse/HV-1900

### CODE REVIEW

File: engine/src/test/java17/org/hibernate/validator/test/constraints/records/RecordConstrainedTest.java [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+		try {
+			new MethodValidationRecord( 50 );
+		}
+		catch (ConstraintViolationException e) {
+			assertThat( e.getConstraintViolations() ).containsOnlyViolations( violationOf( Positive.class ) );
+		}
+	}
+
+	@Test
+	public void byteBuddyRecordTest() {
```
Comment (jrenaat) [inline engine/src/test/java17/org/hibernate/validator/test/constraints/records/RecordConstrainedTest.java:160]: I added this test, even if I didn't actually manage to get it to work; perhaps I'm not adding the validation annotation to the record class correctly, or in order for this 'hot swap' to work we would need a java agent?

**Review by gsmet — CHANGES_REQUESTED** [review]
Looks good, thanks. I added a few comments inline.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
I merged this, thanks. Gonna work on pushing it to the other branches now.

**Comment by ferdinand-swoboda** [comment]
@gsmet Can you give an indication regarding when this change will be released 🙂 ? Thanks

**Comment by Katalystical** [comment]
@gsmet @yrodiere Any chance for a new release? :-)

**Comment by gsmet** [comment]
@Katalystical I will push one today.

**Comment by gsmet** [comment]
@Katalystical sorry I got a bit sidetracked. 6.2.4.Final is available on Maven Central now.

**Pull request merged**